src/views/Pools/components/CakeVaultCard/RecentCakeProfitRow.tsx

```diff
@@ -1,26 +1,44 @@
 import React from 'react'
-import BigNumber from 'bignumber.js'
 import { Flex, Text } from '@pancakeswap/uikit'
-import { useTranslation } from 'contexts/Localization'
 import { useWeb3React } from '@web3-react/core'
+import { useTranslation } from 'contexts/Localization'
+import { useGetApiPrice } from 'state/hooks'
+import useGetVaultUserInfo from 'hooks/cakeVault/useGetVaultUserInfo'
+import useGetVaultPublicData from 'hooks/cakeVault/useGetVaultPublicData'
+import { getBalanceNumber } from 'utils/formatBalance'
+import { getCakeAddress } from 'utils/addressHelpers'
+import { convertSharesToCake } from 'views/Pools/helpers'
 import RecentCakeProfitBalance from './RecentCakeProfitBalance'
 
-interface RecentCakeProfitRowProps {
-  cakeAtLastUserAction: BigNumber
-  userShares: BigNumber
-}
-
-const RecentCakeProfitCountdownRow: React.FC<RecentCakeProfitRowProps> = ({ cakeAtLastUserAction, userShares }) => {
+const RecentCakeProfitCountdownRow = () => {
   const { t } = useTranslation()
   const { account } = useWeb3React()
+  const { cakeAtLastUserAction, userShares, lastUserActionTime } = useGetVaultUserInfo()
   const shouldDisplayCakeProfit =
     account && cakeAtLastUserAction && cakeAtLastUserAction.gt(0) && userShares && userShares.gt(0)
 
+  const { pricePerFullShare } = useGetVaultPublicData()
+  const currentSharesAsCake = convertSharesToCake(userShares, pricePerFullShare)
+  const cakeProfit = currentSharesAsCake.cakeAsBigNumber.minus(cakeAtLastUserAction)
+  const cakeToDisplay = cakeProfit.gte(0) ? getBalanceNumber(cakeProfit, 18) : 0
```

Comment:
Can you ping me with the issue you said this addresses. I think this might be happening on predictions.

Reply:
Ah... so... I _thought_ that the issue here was that `react-countup` didn't respect the decimals passed to it when the value is zero... but actually it does.

It's the ` {value && unit && <span>{unit}</span>}` that's problematic - it renders a 0 in the absence of data.

I've fixed it in this PR here https://github.com/ChefHutch/pancake-frontend/pull/11/files#diff-57e517551373772fa3e382de8bd7fbe40db65ed6e02b81608d7874c8fc73741dR29